hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterStateManager.java

```diff
@@ -284,4 +285,38 @@ public String toString() {
         sb.append('}');
         return sb.toString();
     }
+
+    private static final class StateManagerExceptionHandler implements FutureUtil.ExceptionHandler {
+        private final ILogger logger;
+        private Throwable error;
```

Comment:
What about adding a comment about it's usage instead of making it `volatile`? Since, it's written and read by the same thread.